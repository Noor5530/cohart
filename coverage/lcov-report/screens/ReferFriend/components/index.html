
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for screens/ReferFriend/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> screens/ReferFriend/components</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">70.23% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>59/84</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">56.6% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>30/53</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">71.42% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>10/14</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">69.51% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>57/82</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="PhoneEmailRefer.tsx"><a href="PhoneEmailRefer.tsx.html">PhoneEmailRefer.tsx</a></td>
	<td data-value="95.74" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 95%"></div><div class="cover-empty" style="width: 5%"></div></div>
	</td>
	<td data-value="95.74" class="pct high">95.74%</td>
	<td data-value="47" class="abs high">45/47</td>
	<td data-value="86.66" class="pct high">86.66%</td>
	<td data-value="30" class="abs high">26/30</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	<td data-value="95.65" class="pct high">95.65%</td>
	<td data-value="46" class="abs high">44/46</td>
	</tr>

<tr>
	<td class="file low" data-value="PhoneRefer.tsx"><a href="PhoneRefer.tsx.html">PhoneRefer.tsx</a></td>
	<td data-value="37.83" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 37%"></div><div class="cover-empty" style="width: 63%"></div></div>
	</td>
	<td data-value="37.83" class="pct low">37.83%</td>
	<td data-value="37" class="abs low">14/37</td>
	<td data-value="17.39" class="pct low">17.39%</td>
	<td data-value="23" class="abs low">4/23</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="36.11" class="pct low">36.11%</td>
	<td data-value="36" class="abs low">13/36</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Thu Dec 02 2021 12:57:55 GMT+0500 (Pakistan Standard Time)
            </div>
        </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    