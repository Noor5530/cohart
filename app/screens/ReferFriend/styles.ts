import { appFonts } from 'components/text';
import { iconColor, textColor } from 'config/colors';
import { StyleSheet } from 'react-native';
import { heightPercentageToDP as hp, widthPercentageToDP as wp } from 'react-native-responsive-screen';

const style = StyleSheet.create({
    container: {
        height: hp(100),
        width: '100%',
        backgroundColor: textColor,
        alignItems: 'center',
        justifyContent: 'space-between',
    },
    header: {
        width: '100%',
    },
    mainContainer: {
        flex: 1,
        width: '100%',
        alignItems: 'center',
        paddingHorizontal: wp('5'),
    },
    bottomContainer: { flex: 1, width: '100%', alignItems: 'center' },
    button: {
        height: 35,
        width: 140,
        borderRadius: 30,
        borderWidth: 1,
        borderColor: iconColor,
        alignItems: 'center',
        justifyContent: 'center',
    },
    bottonText: {
        fontWeight: '500',
        fontFamily: appFonts.InterRegular,
        fontSize: 13,
    },
    bottomText: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        height: 40,
    },
    cross: {
        width: 20,
        height: 20,
        borderRadius: 10,
        borderWidth: 1,
        alignItems: 'center',
        justifyContent: 'center',
        marginRight: 10,
    },
    label: {
        fontSize: 13,
        color: iconColor,
    },
    reefersText: {
        paddingTop: 25,
        height: 55,
        textAlignVertical: 'top',
        fontSize: 13,
        color: iconColor,
    },
    summaryText: {
        paddingTop: 10,
        fontFamily: appFonts.InterRegular,
        color: iconColor,
        paddingLeft: wp(4),
        width: wp(85),
        fontSize: 15,
    },
    phoneTextInputTextInputStyle: {
        height: 50.42,
        padding: 0,
        margin: 0,
        width: '90%',
        color: iconColor,
        fontFamily: appFonts.InterRegular,
        fontSize: wp('5'),
        fontWeight: 'bold',
    },
    phoneTextContainerStyle: {
        width: '90%',
        margin: 0,
        backgroundColor: textColor,
        height: 50.42,
    },
    phoneCountryContainerStyle: {
        borderWidth: 2,
        borderColor: iconColor,
        borderRadius: 10,
        paddingHorizontal: 10,
        backgroundColor: textColor,
    },
    phoneCodeTextStyle: {
        margin: 0,
        color: iconColor,
        fontFamily: appFonts.InterRegular,
        fontSize: wp('5'),
        fontWeight: 'bold',
    },
    phoneContainerStyle: {
        padding: 0,
        margin: 0,
        width: '100%',
        alignItems: 'flex-start',
        justifyContent: 'flex-start',
        height: 50.42,
        borderColor: iconColor,
        borderRadius: 14.4067,
        paddingLeft: 20,
        backgroundColor: textColor,
    },
    buttonContainer: {
        flex: 1,
        width: '100%',
        alignItems: 'center',
        paddingHorizontal: wp('5'),
        marginTop: hp('15'),
    },
    buttonStyles: {
        height: hp('5'),
        width: wp('90'),
        alignItems: 'center',
        justifyContent: 'center',
        borderWidth: 0.5,
        borderRadius: 39,
        marginVertical: hp('1'),
        // paddingVertical:5
    },
    buttonTextStyle: {
        fontWeight: '700',
        color: iconColor,
        textTransform: 'uppercase',
    },
});
export default style;
