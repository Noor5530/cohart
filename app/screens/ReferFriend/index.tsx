import { useIsFocused } from "@react-navigation/native";
import StatusBar from "components/CustomStatusBar";
import Header from "components/Header";
import HeaderHeading from "components/HeaderHeading";
import { ReferFriendIcon } from "components/Icons";
import { appFonts } from "components/text";
import { iconColor, textColor } from "config/colors";
import { isIphone } from "lib/isIphone";
import React, { useEffect, useState } from "react";
import { Text, View } from "react-native";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";

import PhoneEmailRefer from "./components/PhoneEmailRefer";
import PhoneRefer from "./components/PhoneRefer";
import styles from "./styles";

interface IProps {
  route: {
    params: {
      showDraw: boolean
    }
  }
}

export default function ReferFriend ({route}: IProps) {
  const isFocused = useIsFocused();
  const [referState, setComponentState] = useState("main");

   useEffect(()=>  {

    if (isFocused) {
      clickOnBackButton()
    }

   }, [isFocused]);

   const clickOnPhone = () => {
    setComponentState("phone")
   }
   const clickOnEmail = () => {
    setComponentState("email")
   }
   const clickOnBackButton = () => {
    setComponentState("main")
   }
return (
    <>
      <StatusBar backgroundColor={textColor} barStyle="dark-content" />

      <View style={styles.container}>
        <View style={styles.header}>
          <Header
            style={{
              paddingTop: isIphone(),
              borderBottomColor: iconColor,
              borderBottomWidth: 1.5,
            }}
            textStyle={{
              color: iconColor,
              fontFamily: appFonts.AktivGroteskEx_Trial_Bd,
            }}
            back={true}
            referBack={referState == "main" ? false : true}
            color={iconColor}
            referBackPress={clickOnBackButton}
          />
          <HeaderHeading
            drawer={route?.params?.showDraw}
            title={"Refer \na Friend"}
            renderIcon={
              <View style={{ top: 10, left: -30 }}>
                <ReferFriendIcon
                  color="rgba(0,0,0,0.5)"
                  width={wp("18")}
                  height={wp("18")}
                />
              </View>
            }
          />
        </View>
        <View key="referFriend" style={{ flex: 1, width: wp(100) }}>
          {referState == "main" && (
            <View style={{ flex: 1 }}
          >
              <Text
                style={[
                  styles.summaryText,
                  {
                    color: "black",
                  },
                ]}
              >
                How would you like to refer your friend?
              </Text>
              <View style={styles.buttonContainer}>
                <View style={styles.buttonStyles}>
                <Text
                    testID="clickOnPhone"
                    onPress={clickOnPhone}
                    style={styles.buttonTextStyle}
                  >
                    Via phone number only
                  </Text>
                </View>
                <View style={[styles.buttonStyles, { marginTop: 25 }]}>
                <Text
                              testID="clickOnEmail"
             onPress={clickOnEmail}
                    style={styles.buttonTextStyle}
                  >
                    via phone number & email address
                  </Text>
                </View>
              </View>
            </View>
          )}
          {referState == "phone" && (
            <PhoneRefer setReferState={setComponentState} />
          )}
          {referState == "email" && (
            <PhoneEmailRefer setReferState={setComponentState} />
          )}
        </View>
      </View>
    </>
  );
}
