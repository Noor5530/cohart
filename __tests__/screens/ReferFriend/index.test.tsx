import "react-native";
import React from "react";
import ReferFriend from "../../../app/screens/ReferFriend";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react-native";

jest.useFakeTimers();

test("render correctly referFriends", () => {
  let component = renderer.create(<ReferFriend />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("run email flow  referFriends", () => {

  let instanceScreen = render(<ReferFriend />);
  const firstFlow = instanceScreen.getByTestId("clickOnPhone");
  fireEvent.press(firstFlow);

  instanceScreen = render(<ReferFriend />);
  const secondFLow = instanceScreen.getByTestId("clickOnEmail");

  fireEvent.press(secondFLow);

  expect(firstFlow).not.toBeNull();
});
test("run phone flow referFriends", () => {
  let instanceScreen = render(<ReferFriend />);
  const secondFLow = instanceScreen.getByTestId("clickOnEmail");

  fireEvent.press(secondFLow);
  expect(secondFLow).not.toBeNull();
});
