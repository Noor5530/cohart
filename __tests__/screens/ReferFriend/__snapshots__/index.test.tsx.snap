// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly referFriends 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "height": undefined,
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomColor": "#000000",
          "borderBottomWidth": 1.5,
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "space-between",
          "paddingHorizontal": 20,
          "paddingTop": 30,
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "width": 38,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="14"
            bbWidth="21"
            fill="none"
            focusable={false}
            height="14"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 14,
                  "width": 21,
                },
              ]
            }
            vbHeight={14}
            vbWidth={21}
            width="21"
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M6.64474 0.583984L0.433134 6.79559L0.167969 7.06076L0.433134 7.32592L6.64474 13.5375L7.17507 13.0072L1.66787 7.5L21 7.5V6.75L1.53939 6.75L7.17507 1.11431L6.64474 0.583984Z"
                fill={4278190080}
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 20,
              },
              Object {
                "color": "#000000",
                "fontFamily": "AktivGroteskExTrial-Bold",
              },
              Object {
                "fontFamily": "AktivGroteskExTrial-Bold",
              },
            ]
          }
        >
          COHART
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="19"
          bbWidth="16"
          fill="none"
          focusable={false}
          height="19"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 19,
                "width": 16,
              },
            ]
          }
          vbHeight={19}
          vbWidth={16}
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M7.58726 1.75C8.01448 1.75 8.36081 2.09633 8.36081 2.52355C8.36081 3.03874 8.66732 3.59741 9.26471 3.7987C11.462 4.53906 13.0903 6.71461 13.0903 9.32316C13.0903 9.36104 13.0899 9.39882 13.0893 9.43649C13.0881 9.50183 13.0931 9.56582 13.1039 9.62789V12.5105C13.1039 12.8394 13.1805 13.1637 13.3275 13.4579L14.1596 15.1219C14.199 15.2007 14.1417 15.2934 14.0536 15.2934H1.11992C1.03185 15.2934 0.974564 15.2007 1.01395 15.1219L1.81191 13.526C1.95899 13.2319 2.03556 12.9075 2.03556 12.5786V9.75294C2.06936 9.64853 2.08682 9.53692 2.08501 9.42124C2.0845 9.38862 2.08424 9.35593 2.08424 9.32316C2.08424 6.71462 3.71252 4.53907 5.90981 3.7987C6.5072 3.59741 6.81371 3.03874 6.81371 2.52355C6.81371 2.09633 7.16004 1.75 7.58726 1.75ZM9.58401 2.85105C9.44664 2.80476 9.36081 2.66851 9.36081 2.52355C9.36081 1.54405 8.56676 0.75 7.58726 0.75C6.60775 0.75 5.81371 1.54405 5.81371 2.52355C5.81371 2.66851 5.72787 2.80476 5.5905 2.85105C2.97586 3.73204 1.08424 6.29665 1.08424 9.32316C1.08424 9.36116 1.08454 9.39908 1.08513 9.43693C1.08535 9.45079 1.07419 9.46217 1.06033 9.46217C1.04665 9.46217 1.03556 9.47326 1.03556 9.48694V12.5786C1.03556 12.7523 0.995133 12.9235 0.91748 13.0788L0.119523 14.6747C-0.252313 15.4184 0.288463 16.2934 1.11992 16.2934H5.15932C5.40744 17.4128 6.406 18.25 7.6001 18.25C8.79419 18.25 9.79276 17.4128 10.0409 16.2934H14.0536C14.8851 16.2934 15.4258 15.4184 15.054 14.6747L14.222 13.0107C14.1443 12.8554 14.1039 12.6841 14.1039 12.5105V9.46957C14.1039 9.46549 14.1006 9.46217 14.0965 9.46217C14.0923 9.46217 14.089 9.45878 14.0891 9.45464C14.0899 9.41091 14.0903 9.36709 14.0903 9.32316C14.0903 6.29665 12.1987 3.73203 9.58401 2.85105ZM8.99864 16.2934H6.20155C6.41923 16.8532 6.96332 17.25 7.6001 17.25C8.23687 17.25 8.78097 16.8532 8.99864 16.2934Z"
              fill={4278190080}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "borderBottomWidth": 2,
          "borderColor": "#000000",
          "flexDirection": "row",
          "paddingTop": 35,
          "width": "90%",
        }
      }
    >
      <Text
        numberOfLines={3}
        style={
          Object {
            "fontFamily": "Inter-Bold",
            "fontSize": undefined,
            "maxWidth": "90%",
            "textAlign": "left",
          }
        }
      >
        Refer 
a Friend
      </Text>
      <View
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "left": -30,
              "top": 10,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="37"
            bbWidth="34"
            fill="none"
            focusable={false}
            height="37"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 37,
                  "width": 34,
                },
              ]
            }
            vbHeight={34}
            vbWidth={37}
            width="34"
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M16.6407 29.5888C26.6842 29.5888 34.5069 25.3791 34.5069 20.1362C34.5069 14.8928 25.798 10.2402 16.1238 10.2402C6.44907 10.2402 1 12.8991 1 15.4099C1 17.9207 6.59717 20.1362 16.1238 20.1362C25.6505 20.1362 34.5069 16.148 34.5069 10.6098C34.5069 5.07079 24.967 1 16.382 1"
                propList={
                  Array [
                    "stroke",
                  ]
                }
                stroke={2147483648}
              />
              <RNSVGPath
                d="M20.6046 32.2462L16.543 29.5878L20.6046 27.0332"
                propList={
                  Array [
                    "stroke",
                  ]
                }
                stroke={2147483648}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "width": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Inter-Regular",
              "fontSize": 15,
              "paddingLeft": undefined,
              "paddingTop": 10,
              "width": undefined,
            },
            Object {
              "color": "black",
            },
          ]
        }
      >
        How would you like to refer your friend?
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "marginTop": undefined,
            "paddingHorizontal": undefined,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 39,
              "borderWidth": 0.5,
              "height": undefined,
              "justifyContent": "center",
              "marginVertical": undefined,
              "width": undefined,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#000000",
                "fontWeight": "700",
                "textTransform": "uppercase",
              }
            }
            testID="clickOnPhone"
          >
            Via phone number only
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 39,
                "borderWidth": 0.5,
                "height": undefined,
                "justifyContent": "center",
                "marginVertical": undefined,
                "width": undefined,
              },
              Object {
                "marginTop": 25,
              },
            ]
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#000000",
                "fontWeight": "700",
                "textTransform": "uppercase",
              }
            }
            testID="clickOnEmail"
          >
            via phone number & email address
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
